src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+  get_grid(t.trian)
+end
+
+function Geometry.get_glue(t::RefinedTriangulation,::Val{d}) where d
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::RefinedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return RefinedTriangulation(v,t.model)
+end
+
+# Wrap constructors for RefinedDiscreteModel
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::RefinedTriangulation,args...;kwargs...)
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  smodel = get_refined_model(strian)
+  tmodel = get_background_model(ttrian)
+  a = get_model(smodel) === tmodel  # It is fundamentally the same model
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  return is_change_possible(ttrian,strian)
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_model(smodel) === get_parent(tmodel)
+  b = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  a || b ? ttrian : strian
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return strian
+end
+
+function Geometry.best_target(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return ttrian
+end
+
+
+"""
+  Given a RefinedTriangulation and a CellField defined on the parent(coarse) mesh, 
+  returns an equivalent CellField on the fine mesh.
+"""
+function change_domain_c2f(f_coarse, ftrian::RefinedTriangulation{Dc,Dp}) where {Dc,Dp}
```

Comment:
If am not wrong, I think that both `change_domain_c2f` (and thus `change_domain` below) only make sense if the `DomainStyle` of `f_coarse` is `ReferenceDomain()`. 

If this is correct, I would implement the `change_domain` version for the particular case in which `DomainStyle==ReferenceDomain()` and leave the one for which `DomainStyle==PhysicalDomain()` 
as `@nonimplemented`

Reply:
Resolved by [0e3c6eb](https://github.com/gridap/Gridap.jl/pull/838/commits/0e3c6eb9cdc1b3dd995a65f7732f940ee853e28b)